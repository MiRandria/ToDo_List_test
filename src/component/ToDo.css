div{
    display: flex;
    margin-top: 20px;
}
.todo{
    box-shadow: 1px 5px 20px 2px rgb(196, 196, 196);
    border-radius: 10px;
    height: 80vh;
    width: 20vw;
    margin-left: 20vw;
    background-color: rgb(243, 239, 243);
    display: inline;
}
.done{
    
    box-shadow: 1px 5px 20px 2px rgb(196, 196, 196);
    border-radius: 10px;
    margin-left: 17vw;
    background-color: rgb(243, 239, 243);
    width: 20vw;
    display: inline;
}
h1{
    margin-left: 7.5vw;
    font-size: x-large;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    
}
p{
    font-size: large;
    margin-left: 3vw;
    margin-top: 9vh;
}
.text{
    height: 4vh;
    width: 18vw;
    margin-top: 20px;
    margin-left: 8px;
    border-radius: 2px;
}
.checkbox{
    width: 10vw;
    margin-top: 20px;
}

