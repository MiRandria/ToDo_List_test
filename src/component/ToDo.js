import React, { useState } from 'react'
import './ToDo.css';

function ToDo() {
  const [value, setValue] = useState("");
  const [stock, setStock] = useState([]);
  const [doneTask, setDoneTask] = useState([]);
  return (
    <>
    <div className='todo'>
        <h1>TODO</h1>
        <input type="text" onChange={(e) => setValue(e.target.value)} onKeyDown={(e) => {
            if (e.key === "Enter") {
                setStock([value,...stock])
            }
        }} placeholder={"Task"} className='text'/>
        {stock.map((task) => {
          return (
            <>
              <input type={"checkbox"} onClick={(e) => {
                const index = stock.indexOf(task); 
                const stockCopie = [...stock];
                stockCopie.splice(index,1)
                setStock(stockCopie) 
                setDoneTask([task,...doneTask])
                }} checked={false} className='checkbox' />
              <span>{task}</span>
            </>
          )
        })}
    </div>
    <div className='done'>
        <h1>DONE</h1>
        {doneTask.map((done) => {
          return (
            <>
              <p>{done}</p>
            </>
          )
        })}
    </div>
    </>
  )
}
export default ToDo;
