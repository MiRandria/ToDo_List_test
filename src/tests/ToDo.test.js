import { render, screen, fireEvent  } from '@testing-library/react';
import React from "react";
import renderer from 'react-test-renderer';
import ToDo from '../component/ToDo';

describe('ToDo component', () => {
  it('Show the ToDo header', () => {
    render(<ToDo/>);
    expect(screen.getByText("TODO")).toBeInTheDocument();
  });
  
  it('Show the Done header', () => {
    render(<ToDo/>);
    expect(screen.getByText("DONE")).toBeInTheDocument();
  });
  
  it('Show the ToDo list', () => {
    render(<ToDo />);
    const input = screen.getByPlaceholderText('Task');
    fireEvent.change(input, { target: { value: 'First task' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 13 });
    expect(screen.getByText('First task')).toBeInTheDocument();
  });

  it('Add task in ToDo list', () => {
    render(<ToDo />);
    const input = screen.getByPlaceholderText('Task');
    fireEvent.change(input, { target: { value: 'First task' } });
    fireEvent.keyDown(input, { key: 'Enter', code: 13 });
    const task = screen.getByText('First task');
    expect(task).toBeInTheDocument();
  });

  it("Show a completed task in the list of done", () => {
    render(<ToDo />);
    const input = screen.getByPlaceholderText("Task");
    fireEvent.change(input, { target: { value: "First task" } });
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    const checkbox = screen.getByText("First task");
    fireEvent.click(checkbox);
    expect(screen.getByText("First task")).toBeInTheDocument();
  });
  
  it("Displays several completed tasks in the list of done", () => {
    render(<ToDo />);
    const input = screen.getByPlaceholderText("Task");
    fireEvent.change(input, { target: { value: "First task" } });
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    fireEvent.change(input, { target: { value: "Second task" } });
    fireEvent.keyDown(input, { key: "Enter", code: 13 });
    const checkbox1 = screen.getByText("First task");
    fireEvent.click(checkbox1);
    expect(screen.getByText("First task")).toBeInTheDocument();
    const checkbox2 = screen.getByText("Second task");
    fireEvent.click(checkbox2);
    expect(screen.getByText("Second task")).toBeInTheDocument();
  });

  it('Snapshots', () => {
    const tree = renderer
      .create(<ToDo/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});