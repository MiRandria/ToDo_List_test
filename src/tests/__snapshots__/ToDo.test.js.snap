// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo component Snapshots 1`] = `
Array [
  <div
    className="todo"
  >
    <h1>
      TODO
    </h1>
    <input
      className="text"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Task"
      type="text"
    />
  </div>,
  <div
    className="done"
  >
    <h1>
      DONE
    </h1>
  </div>,
]
`;
